app/js/actions/jobs.js

```diff
@@ -1,13 +1,119 @@
+            if (!response.ok) {
+                throw Error(response.statusText);
+            }
+            return response.json();
+        })
+        .then((json) => {
+            const completed = [];
+            if (json.completed.length !== 0) {
+                for (const job of json.completed) {
+                    if (Object.keys(job).length !== 0) {
+                        dispatch(actions.jobCompleted(job));
+                        completed.push(job.id);
+                    }
+                }
+            }
+            return completed;
+        })
+        .then((completed) => {
+            if (completed.length !== 0) {
+                dispatch(actions.refreshArtifacts());
+                for (const id of completed) {
+                    url = `http://${uri.split('/')[0]}${availableApis[0]}jobs/${id}`;
+                    const method = 'DELETE';
+                    const requestTime = Date.now();
+                    const digest = makeB64Digest(secretKey, method, url, requestTime, undefined);
+                    fetch(url, {
+                        method,
+                        headers: new Headers({
+                            Authorization: `HMAC-SHA256 ${digest}`,
+                            'Content-Type': 'application/json',
+                            'X-QIIME-Timestamp': requestTime
+                        }),
+                        body: JSON.stringify({})
+                    });
+                }
+            }
+        })
+        .then(() => {
+            const { jobs: { activeJobs } } = getState();
+            if (!activeJobs.length) {
+                clearInterval(jobInterval);
+            }
+        });
+    };
+};
+
+export const startJob = (workflow, data) => {
+    const jobData = {};
+    for (const [key, value] of data.entries()) {
+        jobData[key] = value;
+    }
+
+    return (dispatch, getState) => {
+        const { connection: { uri, availableApis, secretKey } } = getState();
+        const url = `http://${uri.split('/')[0]}${availableApis[0]}${workflow.jobUri}`;
+        const method = 'POST';
+        const requestTime = Date.now();
+        const body = JSON.stringify({
+            jobData
+        });
+        const digest = makeB64Digest(secretKey, method, url, requestTime, body);
+
+        fetch(url, {
+            method,
+            headers: new Headers({
+                Authorization: `HMAC-SHA256 ${digest}`,
+                'Content-Type': 'application/json',
+                'X-QIIME-Timestamp': requestTime
+            }),
+            body
+        })
+        .then((response) => {
+            if (!response.ok) {
+                throw Error(response.statusText);
+            }
+            return response.json();
+        })
+        .then((json) => {
+            if (json.success) {
+                dispatch(actions.newActiveJob(workflow, json.job));
+                jobInterval = setInterval(() => dispatch(actions.pollJobStatus()), 1000);
```

Comment:
Yeah, SSE would be a nice thing to add. The polling just ended up fitting best with the blocking code that was already written. I'll definitely dig into switching it over, as that would be nice to have for all aspects of the app, plugins possibly being able to be pushed to the UI without restarting it, etc...